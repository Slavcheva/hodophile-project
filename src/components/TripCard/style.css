
div .tripCard-wrapper {
    width: 50%;
    margin: 0 auto;
    padding: 3em 0;
    text-align: center;
    border-radius: 1em;
}

div .tripCard-wrapper p {
    display: flex;
    flex-wrap: wrap;
    justify-content: center;
}

div .welcome {
    padding: 10px;
    box-shadow: 0 5px 10px 5px #cecece;
}

.tripCard{
    padding: 10px;
    box-shadow: 0 5px 10px 5px #cecece;
    font: 30px/1.5 'Cormorant', cursive;
}

img {
    max-width: 300px;
    height: auto;
    margin: 0 auto;
}
