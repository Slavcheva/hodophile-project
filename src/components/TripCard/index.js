import React from 'react';
import './style.css'

function TripCard({imageUrl, imageAlt, destination, children, author, _id}) {

    return <div className='tripCard'>
        <h3>{destination}</h3>
        <div className="card-img">
            <img src={imageUrl} alt={imageAlt}/>
        </div>
        <div className="card-content">
            <p>{children}</p>
            <span>
            <small>Author: </small>
                {author}
            </span>
        </div>
    </div>;
}

export default TripCard;