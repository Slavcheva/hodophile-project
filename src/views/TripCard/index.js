import React from 'react';
import './style.css'

function TripCard({imageUrl, imageAlt, destination, children, author, _id}) {

    return <div className="tripCard-wrapper">

        <div className='tripCard'>
            <h5>{destination}</h5>
            <img src={imageUrl} alt={imageAlt}/>
            <p className="description">{children}</p>
            <span>
            <small>Author: </small>
                {author}
            </span>
        </div>
    </div>;
}

export default TripCard;