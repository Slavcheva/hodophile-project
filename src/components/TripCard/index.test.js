import React from "react";
import renderer from 'react-test-renderer'
import TripCard from "./index";

jest.mock("react-router-dom", () => ({
    Link: "Link"
}));

describe('TripCard component', () => {
    test('should render correctly', () => {
        const component = renderer.create(
            <TripCard/>
        )
        const tree = component.toJSON()
        expect(tree).toMatchSnapshot()
    })
})