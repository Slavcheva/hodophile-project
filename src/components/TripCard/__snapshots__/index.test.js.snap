// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TripCard component should render correctly 1`] = `
<div
  className="tripCard-wrapper"
>
  <div
    className="tripCard"
  >
    <h3 />
    <img />
    <p
      className="description"
    />
    <span>
      <small>
        Author: 
      </small>
    </span>
  </div>
</div>
`;
